### DISCUSSION TITLE
Rate limiting and/or fail2ban nginx?

### DISCUSSION THREAD

**Discussion by ribtoks** [plausible/analytics#4527] [Self-Hosted Support]
Hi

Is there any best practice of protecting nginx-based deployment from DDoS using rate limiting and/or fail2ban?

Does it actually negatively affect inound events processing?

[nginx rate limiting docs](https://blog.nginx.org/blog/rate-limiting-nginx)

**Comment by ruslandoga** [comment]
👋 @ribtoks 

There is some rate-limiting built into Plausible: https://github.com/search?q=repo%3Aplausible%2Fanalytics%20ratelimit&type=code - for auth, api, and some other endpoints.

`/api/events` rate-limit can be configured per site via `ingest_rate_limit_threshold` column  in `sites` table in `plausible_db`
By default it means the number of requests allowed in per [60 seconds](https://github.com/plausible/analytics/blob/059b5e0cdd164069b153c79b015919ddb7a46aec/lib/plausible/site.ex#L25) and is not set, meaning no rate limiting applied to `/api/events`.

I don't have experience with NGINX rate-limiting so won't be able to comment on that. It would be interesting to try it out though!

**Reply by ribtoks** [reply]
Hey @ruslandoga . Thank you for fast reply!

Just to confirm if I understood correctly (also, after looking into the code): 
1. by default each site has no rate limiting configured? (`ingest_rate_limit_threshold` is `NULL` by default)
2. the only way to tweak `ingest_rate_limit_threshold` is via a query in Postgres, no other config/UI is available at this time?

**Reply by ruslandoga** [reply]
yes and yes :)

**Reply by ribtoks** [reply]
Thank you! Would be nice if it was possible to set a default inject rate limit somewhere in the general Plausible configuration (could be empty by default for backward compatibility). This way it would be a bit more fool-proof.
